docs/StandardLibrary.md

```diff
@@ -0,0 +1,334 @@
+# Flux Standard Library Design
+
+The Flux standard library will define the set of functions, constants, and other
+values that are available out of the box with Flux, providing the functionality that is
+most frequently used.
+
+Some functions in the standard library will be available without qualifying
+them with a namespace.  These will be the most common functions, to help Flux
+code be more concise.  Such functions are said to be in the "top level package."
+
+The remaining functions in the standard library will be divided up into
+discrete packages that are organized by topic.
+
+What's presented here is just a departure point for further discussion.
+
+## The Top Level Package
+
+Those functions that provide functionality that is very frequently
+used and should not need to be qualified with a namespace in Flux code.
+(Consider also the concept of "prelude")
+
+### Transformations
+Should some of these be in the `math` package instead?  `pearsonr` for example.
+
+Fundamental transformations:
+- `bottom`
+- `distinct`
+- `filter`
+- `group`
+- `join`
+- `keyValues`
+- `keys`
+- `limit`
+- `map`
+- `pivot`
+- `range`
+- `set`
+- `shift`
+- `sort`
+- `stateCount`
+- `stateDuration`
+- `top`
+- `union`
+- `window`
+- `yield`
+
+
+Schema changing operations:
+- `drop`
+- `duplicate`
+- `keep`
+- `rename`
+
+Aggregate operations:
+- `aggregateWindow`
+- `covariance`
+- `cov`
+- `pearsonr`
+- `count`
+- `integral`
+- `mean`
+- `median`
+- `percentile`
+- `skew`
+- `spread`
+- `stddev`
+- `sum`
+
+Selector operations:
+- `first`
+- `last`
+- `max`
+- `min`
+- `percentile`
+- `median`
+- `sample`
+
+Highest/Lowest
+- `highestMax`
+- `highestAverage`
+- `highestCurrent`
+- `lowestMax`
+- `lowestAverage`
+- `lowestCurrent`
+
+Binning transformations:
+- `histogram`
+- `histogramQuantile`
+- `linearBins`
+- `logarithmicBins`
+
+Other mathematical operations:
+- `cumulativeSum`
+- `derivative`
+- `difference`
+
+## Package `testing`
+- `assertEquals`
+
+Others?
+
+## I/O Packages
+
+Should vendor-specific packages like `influx` be part of the standard library?  
+If Flux is to be used as a stand-alone language independent of InfluxDB, perhaps
+they should not.
+
+### Package `csv`
+- `from` (seems lonely here; could be in top level package?)
+
+### Package `http`
+- `from`
+- `to`
+
+### Package `influx`
+- `buckets`
+- `databases`
+- `fieldsAsCols`
+- `from`
+- `fromLP` [IMPL#70](https://github.com/influxdata/flux/issues/70)
+- `fromJSON`
```

Comment:
As per my comment above, I think it would be nice to separate encoding/decoding from the transport of the data. Something to think about here.

Could you do this?

    http.from(url:"...", decoder: influxdb.line)